starlite/middleware/session/base.py

```diff
@@ -0,0 +1,313 @@
+        """Generate a new session-id, with n=[session_id_bytes][starlite.middle
+        ware.session.base.ServerSideSessionConfig.session_id_bytes] random
+        bytes."""
+        return secrets.token_hex(self.config.session_id_bytes)
+
+    async def store_in_message(
+        self, scope_session: "ScopeSession", message: "Message", connection: ASGIConnection
+    ) -> None:
+        """Stores the necessary information in the outgoing `Message` by
+        setting a cookie containing the session-ID. If the session is empty, a
+        null-cookie will be set. Otherwise, the serialised data will be stored
+        using [set][starlite.middleware.session.base.ServerSideBackend.get],
+        under the current session-id. If no session-ID exists, a new ID will be
+        generated using.
+
+        [generate_session_id].
+
+        Args:
+            scope_session: Current session to store
+            message: Outgoing send-message
+            connection: Originating ASGIConnection containing the scope
+        """
+
+        scope = connection.scope
+        headers = MutableHeaders(scope=message)
+        session_id = connection.cookies.get(self.config.key, self.generate_session_id())
+
+        serialised_data = self.serialise_data(scope_session, scope)
+        await self.set(session_id=session_id, data=serialised_data)
+
+        cookie_params = self.config.dict(
+            exclude_none=True,
+            exclude={"secret", "key"} | set(self.config.__fields__) - set(BaseBackendConfig.__fields__),
+        )
+
+        if scope_session:
+            headers["Set-Cookie"] = Cookie(value=session_id, key=self.config.key, **cookie_params).to_header(header="")
+        else:
+            headers.append(
+                "Set-Cookie",
+                Cookie(value="null", key=self.config.key, expires=0, **cookie_params).to_header(header=""),
+            )
+
+    async def load_from_connection(self, connection: ASGIConnection) -> Dict[str, Any]:
+        """Load session data from a connection and return it as a dictionary to
+        be used in the current application scope.
+
+        The session-ID will be gathered from a cookie with the key set in the
+        [configuration[BaseBackendConfig.key]. If a cookie is found, its value will
+        be used as the session-ID and data associated with this ID will be loaded
+        using [get][starlite.middleware.session.base.ServerSideBackend.get]. If no cookie was found or no data was loaded
+        from the store, this will return an empty dictionary.
+
+        Args:
+            connection: Originating ASGIConnection
+        """
+        session_id = connection.cookies.get(self.config.key)
+        if session_id:
+            data = await self.get(session_id)
+            if isinstance(data, dict):
+                return data
+            if data is not None:
+                return self.deserialise_data(data)
+        return {}
+
+
+B = TypeVar("B", bound=SessionBackend)
```

Comment:
Please move to the top of the file. Also, consider renaming from `B` to `TBackend`.

Reply:
Done